editor/src/core/es-modules/package-manager/module-resolution.ts

```diff
@@ -476,15 +691,75 @@ function resolveModuleAndApplySubstitutions(
   importOrigin: string,
   toImport: string,
 ): FileLookupResult {
-  function lookupFn(path: string[], direct: boolean): FileLookupResult {
-    const nodeModulesResult = nodeModulesFileLookup(nodeModules, path, direct)
-    if (isResolveSuccess(nodeModulesResult)) {
-      return nodeModulesResult
-    } else {
-      return projectContentsFileLookup(projectContents, path, direct)
+  // 1 Check if it's a core module - Node JS only
+  // 2 If path begins with /, start searching from file system root
+  // 3 If path begins with /, ./, ../
+  //   3.1 Attempt to load file
+  //   3.2 Attempt to load directory
+  //   3.3 Throw "not found"
+  // 4 If path begins with #
+  //   4.1 Attempt to load package imports
+  // 5 Attempt to load package self
+  // 6 Attempt to load node modules
+  // 7 Throw "not found"
+
+  function lookupFn(path: string, innerImportOrigin: string): FileLookupResult {
+    if (path.startsWith('/') || path.startsWith('./') || path.startsWith('../')) {
+      const pathForLookup = path.startsWith('/')
+        ? path
+        : `${stripTrailingSlash(innerImportOrigin)}/${path}`
+      const pathParts = getPartsFromPath(pathForLookup)
+      const lastPart = last(pathParts)
+      const projectContentsFileLookupResult = projectContentsFileLookup(projectContents, pathParts)
```

Comment:
I think it would make it easier to compare this to the specification if we kept the function names from there. That would be especially useful later down the line when we have to update this according to a change in the spec.
e.g.:
projectContentsFileLookup -> loadAsFile
projectContentsDirectoryLookup -> loadAsDirectory
packageExportsLookup -> loadPackageSelf
etc.

Reply:
Yeah that's a fair comment. I avoided that since this logic isn't actually loading the result, it is just resolving the file, but it makes it easier to track back to the linked algorithm